### PR TITLE
Update walkthrough to remove MDC

### PR DESCRIPTION
**Pull request by wtrocki**
### Remove MDC section

There is no need to use MDC based configuration.
With @darahayes and @secondsun we going to collaborate on showcase app configuration format that users will need to supply - possibly keeping it similar or even the same as the MDC one. Keeping it the same will mean that showcase related documentation can stay the same.

### Remove code ready 

Code ready is very unstable on demo clusters due to memory and CPU limitations. 
It is hard to support a walkthrough with CodeReady as CodeReady issues will come up more often than we want. Integration walkthrough that are in progresss now by @evanshortis will provide such capabilities. 

## Use single container showcase deployment

We deploy PWA served statically from server.
This simplifies server url configuration and makes entire setup 60% shorter

## Direct access to keyclok for Auth

Instructions provided assuming that developers will be able to get access to some keycloak realm to create both server and client side keycloak clients. There are some small challenges around permissions and the way to acccess keycloak for the moment that will be addressed in integrately layer.

## Showcase demo Sync and Auth only

Other demos were removed. Push will be optional. 
For push we can develop another, very short walkthrough that will have this one as a dependency. 

## Limited documentation

We are in the process of moving documentation to different website. Docs.aerogear will not reflect new way of configuing showcase 

## Open questions (Answered below)

If we remove MDC and Auth SDK from JS-SDK is there any reason to showcase keycloak integration

Since we have webpush would it be easier to use webpush integration (I know it was decided to remove it, but..?

### CODE REVIEW

**Comment by wei-lee** [comment]
Thanks @wtrocki. I think this is PR is still WIP. Just want to add a few things I think we should address in the walkthrough.

1. The focus will be just sync and auth. We are not going to add push for now because we are not going to update UPS version in this release so there is no webpush support yet.
2. It should explain how/where developers can get the configuration for each service for the client app. For example, where developers to get the keycloak configuration, and the url for the sync server.
3. It is ok to get rid of codeready integration. I agree it is causing too many issues at the moment. One reason that we need it is we have to modify the configuration file. But now that requirement is gone, I think it will be great if we can just ask developers to deploy the ionic-showcase app container and provide the configurations via env vars. This will make things a lot easier and cleaner for developers to follow. In order to do this we need to update the showcase app first.

I think if we can cover these in the new version of the WT that will be great. 
Wdyt?

**Comment by wtrocki** [comment]
@darahayes @wei-lee  @craicoverflow 

Ready for review

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 1]
```diff
@@ -6,15 +6,15 @@
 
 = Running the mobile showcase application with {mobile-services-name}
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: I am thinking maybe we should call it "GraphQL showcase application" rather than "mobile". Also I think we should remove `{mobile-services-name}` at the end.

@pwright wdyt?

Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: Awesome idea!

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 2]
```diff
@@ -6,15 +6,15 @@
...
 
-This Solution Pattern demonstrates running the mobile showcase application using the services provided by Mobile Developer Services.
+This Solution Pattern demonstrates running the GraphQL PWA Showcase Application using Red Hat Mobile Developer Services.
 
-It will show you how to:
+Solution Pattern will show you how to:
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: Not sure if this change is required. Also `Pattern` should be in lower case.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 3]
```diff
@@ -6,15 +6,15 @@
...
 
-* Build and run the link:{showcase-app-link}[Ionic mobile showcase app].
-* Deploy the data sync server of the mobile showcase app.
-* Protect the showcase mobile app and the backend service using the Identity Management service.
+* Build and run the link:{showcase-app-link}[Ionic Mobile Showcase app].
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: may be we just call it `GraphQL client app`?

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 4]
```diff
@@ -6,15 +6,15 @@
...
+* Deploy the data sync server of the GraphQL Showcase app.
+* Protect the Showcase GraphQL app and the backend service using the Red Hat SSO.
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: `Showcase` -> `showcase`

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 5]
```diff
@@ -6,15 +6,15 @@
...
 
-It will also introduce you to all features offered by the data sync framework.
+It will also introduce you to all features offered by the DataSync framework.
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: is it right to use `DataSync` here? What do we use in the docs?

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 3 above)
Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: It has push integration so showcase still stands :) plus is nice to know the technology - Ionic as it might be appealing to some people

**Comment by wtrocki** [comment]
Verified up to the keycloak part. 
Logging to realm is very tricky as it requires incognito window.
It is not possible to be login to one keycloak (cluster level) and use other keycloak - keycloak is confused because of the different cookie values. Not sure if we should document that in the walkthough. Entire area begs for improvement that can be done on integrately level.

There is also small inconvenience  when creating client config users need to map keycloak one as SSO gives different keys than library expects (upstream keycloak has valid values)

CC @pwright

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 6]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
+.. You should see the *Settings* page of the client.
+.. Change the `Valid Redirect URIs` to '{route-ionic-showcase-server-host}*'
+.. Change the `Web Origins` to '*'
+.. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Use the `Download` button to save the configuration file to a directory on the computer.
 
-. Navigate to the *Configuration* tab of the *mobile app* in the Mobile Developer Console.
+. Navigate to the project config-maps section {openshift-host}/console/project/{walkthrough-namespace}/browse/config-maps
+. Select `webapp-config`
+. Add new `auth` section in config by utilizing the config that was copied in previous step.
+. In configuration replace `auth-server-url` to `url` and `resource` to `clientId`
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: This can be done in showcase but it is more valuable to show users that this needs to be done for their apps rather than hiding this fact in the showcase implementation.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 7]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
+. Configuration should look as follows
+
+---
+    window.showcaseConfig = {
+      "backend": {
+        "serverUrl": "/graphql",
+        "wsServerUrl": ((window.location.protocol === "https:") ? "wss://" : "ws://") + window.location.hostname + "/graphql"
+      }
+      "auth": {
+        "realm": "your realm",
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: We do not know realm name. Walkthough can assume that this can be master realm and then we can fill those fields here automatically so developers will be able to copy them.
We will have keycloak url, realm and clientId (as we will ask users to create it with a certain name)

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 8]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
-.. When prompted for an *Admin User Password* enter:
-+
-----
-admin
-----
-. Complete the wizard.
-This process also creates a new realm in the {customer-sso-name}.
+. As your admin to create new realm for your user and obtain url to be able to login to the keycloak instance
+. In new realm we going to create two new clients - one for server side and one for the client side application.
+. Open URL obtained from adminstrator
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: This is the most problematic part. Generally eval users will not be able to finish this walkthough without intervence of the admin user. I think @pwright has really good proposal to address this problem and this use case can be used as good example when this is needed. 

Fix for that (in integrately) will be very simple - we need to get some extra realm with users that will be matching the users created on the cluster realm (the same password). This will enable developers to explore user facing keycloak without intervention of the cluster or customer admins.

CC @wei-lee

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 9]
```diff
@@ -120,83 +107,17 @@ It should include the string `+connected to messaging service+`.
...
-----
-{route-ionic-showcase-server-host}
-----
-. Click the *Create* button.
-. Navigate to the Configuration tab of the mobile app in the Mobile Developer Console.
-. Copy and save the `mobile-services.json` file, you will require this file in a later task.
-
 [time=10]
-== Build and run the mobile showcase app in a browser
+== Access Showcase PWA App in a browser
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
== Access the showcase app in a browser
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 10]
```diff
@@ -120,83 +107,17 @@ It should include the string `+connected to messaging service+`.
...
 
 The showcase app demonstrates the key capabilities provided by {mobile-services-name}.
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: probably can replace `{mobile-services-name}` with `AeroGear data sync framework`.
```suggestion
The showcase app demonstrates the key capabilities provided by AeroGear Data Sync framework
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 11]
```diff
@@ -120,83 +107,17 @@ It should include the string `+connected to messaging service+`.
...
-
-. Choose *build* from the command palette menu.
-This installs required dependencies.
-
-. Choose *serve*  from the command palette menu.
-This starts a web server to serve the PWA.
-
-. Click the URL displayed in the *serve* terminal console.
-The showcase app is displayed.
+In this task, we will access PWA application that is hosted by server side container that you can run in your browser.
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: I think we can just remove this line. It's not really necessary.
```suggestion
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 12]
```diff
@@ -120,83 +107,17 @@ It should include the string `+connected to messaging service+`.
...
 
+. Go to link:{route-ionic-showcase-server-host}[GraphQL playground, window="_blank"]
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: is it actually the playground?
```suggestion
. Go to link:{route-ionic-showcase-server-host}[Showcase app, window="_blank"]
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 13]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
-admin
-----
-.. When prompted for an *Admin User Password* enter:
-+
-----
-admin
-----
-. Complete the wizard.
-This process also creates a new realm in the {customer-sso-name}.
+. As your admin to create new realm for your user and obtain url to be able to login to the keycloak instance
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
. Make sure you have access to a realm in the user-facing SSO instance. If you don't have access, you can ask your customer-admin user to create one for you. 
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 14]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
+. In new realm we going to create two new clients - one for server side and one for the client side application.
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: is this line part of the instruction? if not we can remove this.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 15]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
+. Open URL obtained from adminstrator
+. Login to keycloak (you will be asked to provide additional details like email and name)
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: we can probably merge the previous step and this one together. Something like:

```
. Get the URL of the keycloak instance and login with the credential provided by the customer-admin user.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 6 above)
Comment (wei-lee) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: is there any particular reason that we ask developer to modify the downloaded config for the showcase app?

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 16]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
 
 === Add Identity Management service to the client app (PWA) and create a realm for users
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: actually if you read the title here, I think this section is just about creating the realm. The next section is actually about creating the clients. So we probably should move some of the content in this section to the next one, or create a different section to cover how to get KC config for the client app.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 14 above)
Comment (wei-lee) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: Also the steps from here doesn't really belong in this section: it's about getting the KC client config for the client app.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 17]
```diff
@@ -218,46 +139,47 @@ In this task, you use the Identity Management service to protect both the client
...
-****
-Is *Identity Management* row listed the `Bound Services` section for *showcase-app*?
-****
-
-[type=verificationFail]
-****
-Verify that you followed each step in the procedure above.  If you are still having issues, contact your administrator.
-****
 
 === Configure Identity Management service for the showcase client and server applications
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: we might want to change the title here as we only cover the server part.
```suggestion
=== Configure Identity Management service for the showcase client applications
```

**Review by wei-lee — CHANGES_REQUESTED** [review]
I have left some comments. We also should remove all the references to `{mobile-services-name}` and replace with something else.

@wtrocki  I am happy to work with you on this together. Are you ok if I push some updates to this branch as well?

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 6 above)
Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: it is not for showcase app itself. It is that SSO adapter for js require it (even in their docs).

Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter :)

The format that SSO provides is old because I think we use some old version there.

**Comment by wtrocki** [comment]
Addressed almost all comments. @wei-lee around sections - I would love to avoid communication issue. If you can edit this PR directly and drop the changes that will be awesome. 

I seen it as global top level title with client and server subtitles but happy to go with anything

**Comment by wtrocki** [comment]
Also sorry for leaving in such rough state

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 8 above)
Comment (pwright) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: @wtrocki  are you proposing a change to the integreatly installer? If so, we need that to happen before this walkthrough works, right?

Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: @pwright I'm just flagging problem in walkthough - technically people might be able to get access to admin and create realm thenselves etc. so it might not be that kind of issue but it is possible area to improve. I just suggested possible way but maybe there is another way to do it.

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 14 above)
Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc]: Got it cleared based on my understanding but feel free to rebuild it

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 18]
```diff
@@ -25,48 +25,31 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
-****
-
-[type=walkthroughResource,serviceName=codeready]
-.CodeReady Workspaces
+.{data-sync-name} Documentation
 ****
-* link:{che-url}[Console, window="_blank"]
-* link:https://developers.redhat.com/products/codeready-workspaces/overview/[Overview, window="_blank"]
-* link:https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces[Documentation, window="_blank"]
+* link:https://github.com/aerogear/voyager-server/tree/master/doc[Getting Started with {data-sync-name}, window="_blank"]
```
Comment (wei-lee) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc:32]: @pwright are we going to have a link of the data sync docs on customer portal, or it is ok if we joint point to the upstream docs for now?

**Comment by wei-lee** [comment]
@wtrocki I pushed a new update to mainly replace `mobile` with `data sync`, and some other minor fixes. If you can take a look that will be great!

File: walkthroughs/1-ionic-showcase/walkthrough.adoc (hunk 18 above)
Comment (wtrocki) [reply walkthroughs/1-ionic-showcase/walkthrough.adoc:32]: Upstream docs will not exist. I think we agreed with Paul that docs in voyager server will feed directly into access.redhat.com. What it means that we probably might be better with previos link.

@darahayes to confirm

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 19]
```diff
@@ -131,32 +127,32 @@ Verify that you followed each step in the procedure above.  If you are still hav
 ****
 
 [time=15]
-== Protect the GraphQL showcase app using the Red Hat SSO
+== Protect the showcase app using the Red Hat SSO
 
-In this task, you use the Red Hat SSO to protect both the client (PWA) and the data sync server.
+In this task, we will protect both the frontend and the backend of the {data-sync-showcase-app} with the {customer-sso-name}.
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc:132]: I'm not big fan of the `protect` word as it is not universally translating to `add authz`. Not big issue

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 20]
```diff
@@ -212,74 +127,69 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+        "wsServerUrl": ((window.location.protocol === "https:") ? "wss://" : "ws://") + window.location.hostname + "/graphql"
+      }
+      "auth": {
+        "realm": "your realm",
+        "url": "https://your-server/auth",
+        "clientId": "your-client"
+      }
+    };
+---
+
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
+
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 21]
```diff
@@ -212,74 +127,69 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Copy the content or use the `Download` button to save the configuration file.
+
+Next we need to update the configuration of the frontend app to secure it:
+
+. Navigate to the link:{openshift-host}/console/project/{walkthrough-namespace}/browse/config-maps[Config Maps page of the OpenShift console].
+. Select the config map that is called `webapp-config`, and edit it.
+. Add a new `auth` section by pasting the content that was copied in the previous step.
+.. You should then replace `auth-server-url` to `url` and `resource` to `clientId`.
+. The content of the config map should look as follows:
+
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
+
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 22]
```diff
@@ -212,74 +127,69 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
-. Log in using the following for both username and password:
-+
-----
-admin
-----
-
-. Create a new client for the showcase data sync server:
+. Make sure you have access to a realm in the {customer-sso-name}. If you don't, you can ask your customer-admin user to create one for you. 
+. Login to the realm in the {customer-sso-name}.
+. Create a new client for the frondend:
```
Comment (wtrocki) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
. Create a new client for the frontend app:
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 23]
```diff
@@ -6,15 +6,15 @@
...
 
-* Build and run the link:{showcase-app-link}[Ionic mobile showcase app].
-* Deploy the data sync server of the mobile showcase app.
-* Protect the showcase mobile app and the backend service using the Identity Management service.
+* Build and run the link:{showcase-app-link}[Ionic based GraphQL Client app].
+* Deploy the data sync server of the GraphQL Showcase app.
+* Protect the showcase GraphQL app and the backend service using the Red Hat SSO.
 
-It will also introduce you to all features offered by the data sync framework.
+It will also introduce you to all features offered by the AeroGear Data Sync framework.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
It will also introduce you to all of the features offered by the AeroGear Data Sync framework.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 24]
```diff
@@ -6,15 +6,15 @@
...
+This Solution pattern demonstrates running the GraphQL PWA Showcase Application using Red Hat Mobile Developer Services.
 
 It will show you how to:
 
-* Build and run the link:{showcase-app-link}[Ionic mobile showcase app].
-* Deploy the data sync server of the mobile showcase app.
-* Protect the showcase mobile app and the backend service using the Identity Management service.
+* Build and run the link:{showcase-app-link}[Ionic based GraphQL Client app].
+* Deploy the data sync server of the GraphQL Showcase app.
+* Protect the showcase GraphQL app and the backend service using the Red Hat SSO.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
* Protect the showcase GraphQL app and the backend service using Red Hat SSO.
```

**Comment by wtrocki** [comment]
Approved!

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 25]
```diff
@@ -120,83 +107,16 @@ It should include the string `+connected to messaging service+`.
...
-. Click the *Create* button.
-. Navigate to the Configuration tab of the mobile app in the Mobile Developer Console.
-. Copy and save the `mobile-services.json` file, you will require this file in a later task.
-
 [time=10]
-== Build and run the mobile showcase app in a browser
+== Access the showcase app in a browser
 
-The showcase app demonstrates the key capabilities provided by {mobile-services-name}.
+The showcase app demonstrates the key capabilities provided by AeroGear Data Sync framework
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
The showcase app demonstrates the key capabilities provided by the AeroGear Data Sync framework
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 26]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
 ****
 
 [time=15]
-== Protect the mobile showcase app using the Identity Management service
-
-In this task, you use the Identity Management service to protect both the client (PWA) and the data sync server.
-
-=== Add Identity Management service to the client app (PWA) and create a realm for users
+== Protect the GraphQL showcase app using the Red Hat SSO
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
== Protect the GraphQL showcase app using Red Hat SSO
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 27]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
-admin
-----
-.. When prompted for an *Admin User Password* enter:
-+
-----
-admin
-----
-. Complete the wizard.
-This process also creates a new realm in the {customer-sso-name}.
+In this task, you use the Red Hat SSO to protect both the client (PWA) and the data sync server.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
In this task, you use Red Hat SSO to protect both the client (PWA) and the data sync server.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 28]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
-Is *Identity Management* row listed the `Bound Services` section for *showcase-app*?
-****
+=== Connect SSO to the client app
 
-[type=verificationFail]
-****
-Verify that you followed each step in the procedure above.  If you are still having issues, contact your administrator.
-****
+. Make sure you have access to a realm in the user-facing SSO instance. If you don't have access, you can ask your customer-admin user to create one for you. 
+. In new realm we going to create two new clients - one for server side and one for the client side application.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: This isn't really a step in the procedure, it's more of a description of what we're going to do. I think you could put this up under the `Connect SSO to the client app` heading but before the bullet points. Also it's got some grammar that native speakers will find strange.

I would maybe reword it to the following.

`In the following steps, we will create a new keycloak realm. In the new realm we will create two keycloak clients. One for the server side, and one for the client side application'

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 29]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+. Open URL obtained from adminstrator
+. Login to keycloak (you will be asked to provide additional details like email and name)
+. When completing login process you should see the main keycloak realm page
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
. You should see the main Keycloak realm page when logged in
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 30]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+. In the right menu choose *Clients* under *Configure* in the menu.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: Is this in the on the right side? Is it not on the left side?

Also you're saying 'in the menu choose x in the menu'

Maybe it could be something like 

```suggestion
. In the menu on the right side, choose *Configure* > *Clients*.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 31]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Click on the *Create* button on the top-right corner to start creating a new client.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
.. Click on the *Create* button on the top-right corner to create a new client.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 32]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Enter `showcase-app-client` for Client ID.  Do not change the rest of the fields.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
.. Enter `showcase-app-client` for Client ID. Do not change any other fields.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 33]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Save the client. You should see the *Settings* page of the client.
+.. You should see the *Settings* page of the client.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]: Duplicate of the line above

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 34]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Change the `Valid Redirect URIs` to '{route-ionic-showcase-server-host}*'
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
.. Change `Valid Redirect URIs` to '{route-ionic-showcase-server-host}*'
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 35]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Change the `Web Origins` to '*'
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
.. Change `Web Origins` to '*'
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 36]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+.. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Use the `Download` button to save the configuration file to a directory on the computer.
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
.. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Use the `Download` button to save the configuration file to a directory on your computer.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 37]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
 
-=== Configure Identity Management service for the showcase client and server applications
+. Navigate to the project config-maps section {openshift-host}/console/project/{walkthrough-namespace}/browse/config-maps
+. Select `webapp-config`
+. Add new `auth` section in config by utilizing the config that was copied in previous step.
+. In configuration replace `auth-server-url` to `url` and `resource` to `clientId`
+. Configuration should look as follows
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
. The configuration should look similar to the one below.
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 38]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
 
-. Navigate to the *Mobile Services* tab.
+---
+    window.showcaseConfig = {
+      "backend": {
+        "serverUrl": "/graphql",
+        "wsServerUrl": ((window.location.protocol === "https:") ? "wss://" : "ws://") + window.location.hostname + "/graphql"
+      }
+      "auth": {
+        "realm": "your realm",
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
        "realm": "<your realm>",
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 39]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+        "url": "https://your-server/auth",
+        "clientId": "your-client"
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
        "clientId": "<your-client>"
```

File: walkthroughs/1-ionic-showcase/walkthrough.adoc [hunk 40]
```diff
@@ -212,57 +132,52 @@ Verify that you followed each step in the procedure above.  If you are still hav
...
+      }
+    };
+---
 
-. Expand the *Identity Management* section.
+. Save and wait for server to redeploy
```
Comment (darahayes) [inline walkthroughs/1-ionic-showcase/walkthrough.adoc]:
```suggestion
. Save and wait for the server to redeploy
```

**Review by wei-lee — APPROVED** [review]
